### PR TITLE
Change the default sampler to ParentOrElse(AlwaysOn)

### PR DESCRIPTION
**Pull request by MitchellDumovic**
Discussed at the Spec SIG this morning 7/28. Resolves https://github.com/open-telemetry/opentelemetry-specification/issues/728

### CODE REVIEW

**Comment by dyladan** [comment]
Because it is the default, you may want to move it to the top of the list. I would also consider adding a new heading "Default Sampler" that specifically calls out that this is the default sampler so it isn't hidden in a bullet list.

**Comment by MitchellDumovic** [comment]
@dyladan I pushed some changes that hopefully address the problem here. I have a slight preference for not leading with `ParentOrElse` as it is a more complex sampler type and the default `ParentOrElse(AlwaysOn)` references both `ParentOrElse` and `AlwaysOn`. 

It felt more right to me to include the default header right next to the documentation of the built-in sampler types rather than adding a new higher level header, but again this is a small preference so happy to split out a separate header as well if you think that is more clear.

File: specification/trace/sdk.md [hunk 1]
```diff
@@ -94,6 +94,8 @@ be displayed on debug pages or in the logs. Example:
 Description MUST NOT change over time and caller can cache the returned value.
 
 ### Built-in samplers
+OpenTelemetry supports a number of built-in samplers to choose from. 
+The default sampler if one is not specified is `ParentOrElse(AlwaysOn)`.
```
Comment (dyladan) [inline specification/trace/sdk.md:98]:
```suggestion
The default sampler if is `ParentOrElse(AlwaysOn)`.
```

**Comment by Oberon00** [comment]
I'm OK with this change, but IMHO it was merged far too fast. #745 suggests to wait two business days, but even without that, I think merging a non-trivial change on the same day it was proposed is unfair.

**Comment by yurishkuro** [comment]
@Oberon00 my bad. Is there some bot we can install that would block merge for a predefined "public comment" period? Similar to how WIP bot blocks the merge.

**Pull request merged**